packages/flutter_tools/lib/src/vmservice.dart

```diff
@@ -784,6 +786,37 @@ class FlutterVmService {
     }
   }
 
+  /// Waits for a signal from the VM service that [extensionName] is registered.
+  ///
+  /// Looks at the list of loaded extensions for first Flutter view, as well as
+  /// the stream of added extensions to avoid races.
+  Future<vm_service.IsolateRef> findExtensionIsolate(String extensionName) async {
+    final vm_service.IsolateRef isolateRef = (await getFlutterViews()).first.uiIsolate;
+    try {
+      await service.streamListen(vm_service.EventStreams.kIsolate);
+    } on vm_service.RPCError {
+      // Do nothing, since the tool is already subscribed.
+    }
+
+    try {
+      await Future.any(<Future<void>>[
+        service.getIsolate(isolateRef.id).then(
+          (vm_service.Isolate isolate) async => isolate.extensionRPCs.contains(extensionName)
+            ? null
+            // Never complete.
+            : Completer<void>().future,
+        ),
+        service.onIsolateEvent.firstWhere(
+          (vm_service.Event event) => event.kind == vm_service.EventKind.kServiceExtensionAdded
+              && event.extensionRPC == extensionName,
+        ),
+      ]);
+      return isolateRef;
+    } finally {
+      await service.streamCancel(vm_service.EventStreams.kIsolate);
```

Comment:
I don't know if this is needed - it seems like it's a common pattern here to *not* cancel streams e.g. https://github.com/flutter/flutter/blob/7672716df06ca58e71dcf7290c2e1286e510bb68/packages/flutter_tools/lib/src/vmservice.dart#L484

I can follow up and add more `streamCancels` elsewhere if that is desired though